tests/ignite/contrib/handlers/test_lr_finder.py

```diff
@@ -323,35 +317,105 @@ def test_lr_suggestion_unexpected_curve(lr_finder, to_save, dummy_engine, datalo
         lr_finder.lr_suggestion()
 
 
-def test_lr_suggestion(lr_finder, to_save, dummy_engine, dataloader):
-    with lr_finder.attach(dummy_engine, to_save) as trainer_with_finder:
-        trainer_with_finder.run(dataloader)
+def test_lr_suggestion_single_param_group(lr_finder):  # , to_save, dummy_engine, dataloader):
+
+    noise = 0.05
+    lr_finder._history["loss"] = torch.linspace(-5.0, 5.0) ** 2 + noise
+    lr_finder._history["lr"] = torch.linspace(0.01, 10)
+
+    # lr_finder.lr_suggestion() is supposed to return a value, but as
+    # we assign loss and lr to tensors, instead of lists, it will return tensors
+    suggested_lr = lr_finder.lr_suggestion()
+
+    # Ignoring the increasing part of the curve in the assertion.
+    assert 0.1 <= suggested_lr.item() <= 6
+
+
+def test_lr_suggestion_multiple_param_groups(lr_finder):
+    import numpy as np
+
+    noise = 0.06
+    lr_finder._history["loss"] = torch.tensor(np.linspace(-5.0, 5, num=50) ** 2 + noise)
+    # 2 param_groups
+    lr_finder._history["lr"] = torch.tensor(np.linspace(0.01, 10, num=100)).reshape(50, 2)
+
+    # lr_finder.lr_suggestion() is supposed to return a list of values,
+    # but as we assign loss and lr to tensors, instead of lists, it will return tensors
+    suggested_lrs = lr_finder.lr_suggestion()
+
+    # Ignoring the increasing part of the curve in the assertion.
+    assert 0.1 <= suggested_lrs[0].item() <= 6
+    assert 0.1 <= suggested_lrs[1].item() <= 6
```

Comment:
Can we make these assertions more precise: `pytest.approx(suggested_lrs[0]) == 0.xyz`